From c4d65e0c2aeaa75999b17e928f9de985f18251bb Mon Sep 17 00:00:00 2001
From: Andiry Xu <jix024@cs.ucsd.edu>
Date: Sat, 4 Feb 2017 01:13:40 -0800
Subject: [PATCH 1/8] Export radix_tree_replace_slot

Signed-off-by: Andiry Xu <jix024@cs.ucsd.edu>
---
 lib/radix-tree.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/radix-tree.c b/lib/radix-tree.c
index 84812a9..72fab49 100644
--- a/lib/radix-tree.c
+++ b/lib/radix-tree.c
@@ -1102,6 +1102,7 @@ void radix_tree_replace_slot(struct radix_tree_root *root,
 {
 	replace_slot(root, NULL, slot, item, true);
 }
+EXPORT_SYMBOL(radix_tree_replace_slot);
 
 /**
  * radix_tree_iter_replace - replace item in a slot
-- 
2.7.4

